anesthetic/samples.py

```diff
@@ -118,9 +118,14 @@ def plot(self, ax, paramname_x, paramname_y=None, *args, **kwargs):
 
         if paramname_y is None or paramname_x == paramname_y:
             xmin, xmax = self._limits(paramname_x)
-            return plot_1d(ax, self[paramname_x].compress(),
-                           xmin=xmin, xmax=xmax,
-                           *args, **kwargs)
+            plot_type_1d = kwargs.pop('plot_type_1d', 'kde')
```

Comment:
I'm also not sure about the introduction of this new keyword argument. It makes a lot of sense for your use-case (i.e. histograms on diagonal, KDEs elsewhere), but isn't very consistent with the other ``MCMCSamples`` functions that call ``MCMCSamples.plot``.

Ideally, what we would have is a single ``plot_type`` kwarg, which can take values 
- ``'kde'``, which in the 2D case would give you kdes on both off-diagonal and diagonals, 
- ``'hist'`` which (when implemented) would give you histograms everywhere. For now, you should put in a ``NotImplementedError`` in the 2d sections of the code when users try to call this
-  A dictionary ``{'1d':'hist', '2d':'kde'}``, which would give your use case

You'd need to have a little more logic in the kwarg parsing that uses ``isinstance`` to check if it's a dictionary

Reply:
A 2D hist is typically used much less, because it doesn't combine with other models/data... That why I wanted 1D hists separate from 2D, but a dictionary might be more elegant.

Reply:
I'm not sure I agree (and indeed corner makes a lot of use of 2D histograms). In general there are quite a few options for histograms/kdes (including plotting both). It's best to try and keep the functionality all inside a single keyword argument if possible.

Reply:
I turned types and plot_type into dictionaries. I think this might make the keywords upper and diagonal obsolete, as one can simply check whether the key '1d' exists in the dictionary (if so then diagonal=True) and whether there is just one or two entries for the key '2d'.

What do you think?